### DISCUSSION TITLE
Changing Compile Commands in cp editor on macOS

### DISCUSSION THREAD

**Discussion by srijaygusain** [cpeditor/cpeditor#1079] [Q&A]
By default the compile command is c++ -Wall in cp editor. I changed my default compiler from Clang to GCC. But still cp editor compiles with Clang. I doubt if the change of compile commands feature is even working?
Any help would be appreciated.

**Comment by ouuan** [comment]
What do you mean by "changed my default compiler from Clang to GCC"? What's your current compile command in CP Editor?

**Reply by srijaygusain** [reply]
Background: I have a macbook. So by nature, it has Clang compiler for C++ in it. However,  Clang does not compile many features like __gcd function and ordered_set. So, I first installed GCC. After that I made it my default compiler.
(Default Compiler: compiler invoked when executing command  "g++ A.cpp")

My current compile command is "g++ -Wall"

On executing a program containing __gcd function , compilation fails...So, I believe that still Clang compiler is being used.

**Comment by ouuan** [comment]
It seems that `__gcd` also works in Clang, but you need to `#include <algorithm>` and use `std::__gcd` (or `using namespaces std;`, etc.)

**Reply by srijaygusain** [reply]
Even I include that header file and use standard namespace...but error prevails

**Reply by ouuan** [reply]
> Even I include that header file and use standard namespace...but error prevails

You can try compiling your code in the terminal first to see whether it's related to CP Editor.

**Reply by srijaygusain** [reply]
Yes ....it works in the terminal.... Even in VS code  it works....only in cp editor it does not work

**Comment by ouuan** [comment]
I don't know what could be the cause, just some guesses:

1. Maybe you are using a tab's specific compile command. You can check the tab's compile command or simply open a new tab.
2. Check again your compile command in CP Editor.
3. Try specifying the path to the compiler in the compile command.
4. I don't use macOS, so maybe it's a macOS-specific problem. But we haven't received reports on it so it's unlikely to be a bug. Maybe it's some macOS configuration, I don't know.

**Reply by srijaygusain** [reply]
Thanks for your contribution

**Comment by coder3101** [comment]
Did you correctly setup gcc as default compiler? On macOS it is notoriously difficult to point c++ to g++

Follow this answer

https://stackoverflow.com/questions/61292460/how-do-i-make-sure-that-my-default-c-c-compiler-is-gcc

**Comment by srijaygusain** [comment]
Actually i followed a tutorial on youtube for this
https://www.youtube.com/watch?v=CZ7Mf7qxbIU&t=75s

**Reply by coder3101** [reply]
So did you put `g++-11 -Wall` as the compile command

**Comment by srijaygusain** [comment]
I have put g++ -Wall as the compile command. Before that, I ran a command "ln -s g++-11  g++" which means call g++-11 when g++ is called.
